docs/src/components/action/InfoCard.tsx

```diff
@@ -0,0 +1,71 @@
+import * as React from 'react';
+import Box from '@mui/material/Box';
+import Paper from '@mui/material/Paper';
+import Typography from '@mui/material/Typography';
+import Link from 'docs/src/modules/components/Link';
+
+interface InfoCardProps {
+  icon: React.ReactNode;
+  title: string;
+  description: string;
+  link?: string;
+  linkable?: boolean;
+}
+
+export default function InfoCard({ icon, title, description, link, linkable }: InfoCardProps) {
+  return (
+    <Paper
+      component={linkable ? Link : 'div'}
+      href={link}
+      noLinkStyle={linkable}
```

Comment:
The `link` and `linkable` props look closely related, maybe we could use just `link`?

```suggestion
      component={Boolean(link) ? Link : 'div'}
      href={link}
      noLinkStyle={Boolean(link)}
```

Reply:
Interesting! Why do wee need this `Boolean` thing? Never seen it before.

Reply:
`Boolean()` casts it into a boolean ([reference](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Boolean#boolean_coercion)), so for example

```js
Boolean(undefined) => false
Boolean('https://some-url') => true
```

For the `noLinkStyle` prop, it ensures that we pass a Boolean type, which might help avoid some edge cases. It's not necessary in the ternary as the casting happens natively, but I prefer to do it explicitly. You can also do it with the `!!` operator

```js
!!undefined => false
!!'https://some-url' => true
```

Reply:
Gotcha, appreciate the explanation! Interestingly, it seems Lint doesn't allow the casting on a ternary, but it does when it's purely boolean (i.e., worked out on `noLinkStyle`). Here's what it displayed:

<img width="280" alt="Screen Shot 2023-09-15 at 13 52 13" src="https://github.com/mui/material-ui/assets/67129314/03efaa5d-6aa3-41be-8848-659d5494b0c6">